app/gameboard_svc.py

```diff
@@ -0,0 +1,127 @@
+                                                             red_op_access=op.access)
+        if not self._detection_exists(blue_op[0], verify_type, link):
+            await self._add_detection_to_operation(op=blue_op[0], link_pid=link.pid, verify_type=verify_type)
+            return op.id, 'Detection successfully added for link in operation: ' + op.name + ' - ' + str(op.start) + '.'
+        return None, 'Detection already exists for link in operation: ' + op.name + ' - ' + str(op.start) + '.'
+
+    async def is_link_verified(self, verify_type, link_id):
+        op, link = await self._find_op_and_link(link_id)
+        blue_op = await self.data_svc.locate('operations', dict(name=op.name+'_'+str(op.id)+self.blue_op_name_modifier))
+        if blue_op and self._detection_exists(blue_op[0], verify_type, link):
+            return True
+        return False
+
+    """PRIVATE"""
+
+    async def _find_op_and_link(self, link_id):
+        operations = await self.get_service('data_svc').locate('operations')
+        for op in operations:
+            for link in op.chain:
+                if str(link_id) == link.unique:
+                    return op, link
+        return None, None
+
+    @staticmethod
+    async def _does_pid_match_any_run_link(links, pid):
+        for link in links:
+            if link.pid == int(pid):
+                return link.id
+        return False
+
+    async def _does_guid_match_any_run_link(self, links, guid):
+        pids = [str(link.pid) for link in links]
+        blue_op_name = self.get_config(prop='op_name', name='response')
+        auto_collect_ops = await self.data_svc.locate('operations', match=dict(name=blue_op_name))
+        for auto_collect in auto_collect_ops:
+            for link in auto_collect.chain:
+                link_pid = await self._is_guid_found_by_link(link, guid, pids)
+                if link_pid:
+                    return await self._does_pid_match_any_run_link(links, link_pid)
+        return False
+
+    @staticmethod
+    async def _is_guid_found_by_link(link, guid, pids):
+        for rel in link.relationships:
+            if rel.source.trait == 'host.process.id' and rel.source.value in pids and rel.target and \
+                    rel.target.trait == 'host.process.guid' and rel.target.value == guid:
+                return rel.source.value
+        return False
+
+    async def _create_detection_operation(self, red_op_name, red_op_id, red_op_access):
+        planner = (await self.get_service('data_svc').locate('planners', match=dict(name='atomic')))[0]
+        adversary_id = self.get_config(prop='adversary', name='gameboard')
+        adversary = (await self.data_svc.locate('adversaries', match=dict(adversary_id=adversary_id)))[0]
+        obj = (await self.data_svc.locate('objectives', match=dict(name='default')))[0]
+        agent = Agent(0, 0, 0, paw='gameboard_detection')
+        access = self.Access.BLUE if red_op_access == self.Access.RED else self.Access.HIDDEN
```

Comment:
consider just leaving this blue operation access up to the caller instead of inferring it from the red operation's settings